src/openms/source/ANALYSIS/MAPMATCHING/TransformationModel.cpp

```diff
@@ -60,5 +63,295 @@ namespace OpenMS
+      x_weight_ = x_weight;
+    }
+    else{
+      std::string x_weight = "";
+      x_weight_ = x_weight;
+    }
+    // weight y values
+    valid_weights = getValidYWeights();
+    if (params.exists("y_weight") && checkValidWeight(params.getValue("y_weight"), valid_weights) && !data.empty())
+    {
+      std::string y_weight = params.getValue("y_weight");
+      y_weight_ = y_weight;
+      for (size_t i = 0; i < data.size(); ++i)
+      {
+        data[i].second = weightDatum(data[i].second,y_weight_);
+      }
+    }    
+    else if (params.exists("y_weight") && checkValidWeight(params.getValue("y_weight"), valid_weights))
+    {
+      std::string y_weight = params.getValue("y_weight");
+      y_weight_ = y_weight;
+    }
+    else{
+      std::string y_weight = "";
+      y_weight_ = y_weight;
+    }
+  }
+  
+  void TransformationModel::unWeightData(TransformationModel::DataPoints& data, const Param& params)
+  {
+    // unweight x values 
+    std::vector<std::string> valid_weights;
+    valid_weights = getValidXWeights();
+    if (params.exists("x_weight") && checkValidWeight(params.getValue("x_weight"), valid_weights) && !data.empty())
+    {
+      std::string x_weight = params.getValue("x_weight");
+      x_weight_ = x_weight;
+      for (size_t i = 0; i < data.size(); ++i)
+      {
+        data[i].first = unWeightDatum(data[i].first,x_weight_);
+      }
+    }
+    else if (params.exists("x_weight") && checkValidWeight(params.getValue("x_weight"), valid_weights))
+    {
+      std::string x_weight = params.getValue("x_weight");
+      x_weight_ = x_weight;
+    }
+    else{
+      std::string x_weight = "";
+      x_weight_ = x_weight;
+    }
+    // unweight y values
+    valid_weights = getValidYWeights();
+    if (params.exists("y_weight") && checkValidWeight(params.getValue("y_weight"), valid_weights) && !data.empty())
+    {
+      std::string y_weight = params.getValue("y_weight");
+      y_weight_ = y_weight;
+      for (size_t i = 0; i < data.size(); ++i)
+      {
+        data[i].second = unWeightDatum(data[i].second,y_weight_);
+      }
+    }    
+    else if (params.exists("y_weight") && checkValidWeight(params.getValue("y_weight"), valid_weights))
+    {
+      std::string y_weight = params.getValue("y_weight");
+      y_weight_ = y_weight;
+    }
+    else{
+      std::string y_weight = "";
+      y_weight_ = y_weight;
+    }
+  }
+
+  bool TransformationModel::checkValidWeight(const std::string& weight, const std::vector<std::string>& valid_weights) const
+  {    
+    bool valid = false;
+    if (std::find(valid_weights.begin(), valid_weights.end(), weight) != valid_weights.end())
+    {
+      valid=true;
+    }
+    else
+    {
+      std::cout << "weight " + weight + " is not supported" << std::endl;
+    }
+    return valid;
+  }
+  
+  std::vector<std::string> TransformationModel::getValidXWeights() const
+  {
+    //std::vector<std::string> valid_weights{"1/x","1/x2","ln(x)",""}; C++ 11
```

Comment:
OK great.  It was a bit of a pain working around some of the missing C++ features.  Are you refering to the function for the variable declaration when you say "static const"?